src/components/App/App.js

```diff
@@ -0,0 +1,353 @@
+import { useState, useEffect } from "react";
+import Header from "../Header/Header";
+import Main from "../Main/Main";
+import Footer from "../Footer/Footer";
+import EditProfileModal from "../EditProfileModal/EditProfileModal";
+import LoginModal from "../LoginModal/Login";
+import RegisterModal from "../RegisterModal/RegisterModal";
+import {
+  getToken,
+  getTopArtistData,
+  getNewReleaseData,
+  getRecommendations,
+} from "../../utils/SpotifyAPI/SpotifyAPI";
+import { useHistory } from "react-router-dom/cjs/react-router-dom";
+import { checkToken, register, signin } from "../../utils/auth";
+import UserDataContext from "../contexts/userDataContext";
+import {
+  editUserProfile,
+  getFavoriteArtists,
+  getFavoriteAlbums,
+  getFavoriteTracks,
+  addFavortieTrack,
+  addFavortieArtist,
+  addFavortieAlbum,
+  deleteFavoriteAlbum,
+  deleteFavoriteTrack,
+  deleteFavoriteArtist,
+} from "../../utils/api";
+import arrow from "../../images/uparrow.png";
+
+function App() {
+  const history = useHistory();
+  const [activeModal, setActiveModal] = useState("");
+  const [authToken, setAuthToken] = useState("");
+  const [topArtistData, setTopArtistData] = useState([]);
+  const [newReleaseData, setNewReleaseData] = useState([]);
+  const [recommendations, setRecommendations] = useState([]);
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [currentUser, setCurrentUser] = useState({});
+  const [favoriteArtists, setFavoriteArtists] = useState([]);
+  const [favoriteTracks, setFavoriteTracks] = useState([]);
+  const [favoriteAlbums, setFavoriteAlbums] = useState([]);
+  const [isVisible, setIsVisible] = useState(false);
+
+  const handleCloseModal = () => {
+    setActiveModal("");
+  };
+
```

Comment:
You can add an `Escape` listener like this to be able to remove it **correctly** when modals are closed:

```jsx
   useEffect(() => {

    if (!activeModal) return; // stop the effect not to add the listener if there is no active modal

    const handleEscClose = (e) => {  // define the function inside useEffect not to lose the reference on rerendering
      if (e.key === "Escape") {
        handleCloseModal();
      }
    };

    document.addEventListener("keydown", handleEscClose);

    return () => {  // don't forget to add a clean up function for removing the listener
      document.removeEventListener("keydown", handleEscClose);
    };
  }, [activeModal]);  // watch activeModal here
```

Here you define function ` handleEscClose` inside the `useEffect` so that it could maintain a **fixed** reference if the component rerenders.
And of course we need to return a `clean up` function to **`remove`** the listener if the component unmounts.
And don’t forget about the **`dependencies array`** in order to watch `activeModal`